### PR TITLE
Workarounds to make constitution executable optionally buildable.

### PR DESCRIPTION
**Pull request by bezirg**
Also I fixed and improved some comments in the constitution package.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -95,6 +95,14 @@ allow-newer:
   , inline-r:containers
   , inline-r:primitive
 
+-- Uncomment the following lines to make cardano-constitution:create-json-envelope buildable:True
```
Comment (ana-pantilie) [inline cabal.project:98]: This is probably a silly question, but is there no way to tell cabal to only build this if the user turns on some flag?

File: cardano-constitution/README.md [hunk 2]
```diff
@@ -193,3 +200,30 @@ the union (Sum) of the ledger guarantees required by all our current implementat
 - `src/Cardano/Constitution/Validator/Common.hs` : common code between the 2 engines
 - `data/*` : contains the JSON configuration files
 - `test/*`: testing code
+- `app/create-json-envelope`: an executable to construct constitution script, ready to be submitted to the chain
+
+## Building the script
+
+We have created an executable `create-json-envelope` to ease the creation of a constitution script.
+This executable is disabled by default; to enable it, uncomment the related lines inside this repo's `cabal.project`:
```
Comment (ana-pantilie) [inline cardano-constitution/README.md:208]: Is there a future plan to move this out of the repo?

Comment (bezirg) [reply cardano-constitution/README.md:208]: Yes

File: cabal.project (hunk 1 above)
Comment (bezirg) [reply cabal.project:98]: Good question, but I don't know how :man_shrugging:

**Pull request merged**